### PR TITLE
Merge in upstream master @ v0.4.0

### PR DESCRIPTION
**Pull request by AndrewFerr**
[v0.4.0](https://github.com/mautrix/signal/releases/tag/v0.4.0) = 90542f770

### CODE REVIEW

File: mautrix_signal/db/upgrade/__init__.py [hunk 1]
```diff
@@ -15,4 +15,5 @@
     # Since we merged the schema update `v08_user_activity`, all upgrades since `8` must be bumped by one version to avoid a clash. Please do this when merging.
     v09_disappearing_messages,
     v10_group_topic,
+    v11_puppet_name_quality,
```
Comment (jaller94) [inline mautrix_signal/db/upgrade/__init__.py:18]: This does not match the filename.

File: mautrix_signal/example-config.yaml [hunk 2]
```diff
@@ -265,6 +304,15 @@ bridge:
...
             m.video: '$sender_displayname sent a video'
             m.location: '$sender_displayname sent a location'
+        # Specify a dedicated relay account. Must be a regular matrix account logged into this bridge
+        # and double puppeting working to auto-accept invites. When this user is invited to a room
+        # it will automatically be set as the relay user. May be overridden with `set-relay` or `unset-relay`
+        relaybot: '@relaybot:example.com'
+
+    # Format for generting URLs from location messages for sending to Signal
+    # Google Maps: 'https://www.google.com/maps/place/{lat},{long}'
+    # OpenStreepMap: 'https://www.openstreetmap.org/?mlat={lat}&mlon={long}'
```
Comment (jaller94) [inline mautrix_signal/example-config.yaml:314]: Found two typos and submitted them upstream:
https://github.com/mautrix/signal/pull/311

**Review by jaller94 — CHANGES_REQUESTED** [review]
The migration isn't working. There's an issue with the filename.

**Review by jaller94 — APPROVED** [review]
Couldn't find any issues, but I hope we test it on staging first.
It's a big change.

**Comment by AndrewFerr** [comment]
Should dependencies be locked? Our mautrix-telegram fork locks them: https://github.com/vector-im/mautrix-telegram/blob/element-master/requirements.txt

**Pull request merged**